internal/page/supporter/enter_reference_number.go

```diff
@@ -0,0 +1,88 @@
+package supporter
+
+import (
+	"net/http"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/sesh"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/validation"
+)
+
+type enterReferenceNumber struct {
+	App    page.AppData
+	Errors validation.List
+	Form   *referenceNumberForm
+}
+
+func EnterReferenceNumber(tmpl template.Template, organisationStore OrganisationStore, sessionStore sesh.Store) page.Handler {
+	return func(appData page.AppData, w http.ResponseWriter, r *http.Request) error {
+		data := &enterReferenceNumber{
+			App: appData,
+			Form: &referenceNumberForm{
+				Label: "referenceNumber",
+			},
+		}
+
+		if r.Method == http.MethodPost {
+			data.Form = readReferenceNumberForm(r, "referenceNumber")
+			data.Errors = data.Form.Validate()
+
+			if data.Errors.None() {
+				invite, err := organisationStore.InvitedMember(r.Context())
+				if err != nil {
+					return err
+				}
+
+				if invite.ReferenceNumber != data.Form.ReferenceNumber {
+					data.Errors.Add("reference-number", validation.CustomError{Label: "incorrectReferenceNumber"})
+					return tmpl(w, data)
+				}
+
+				if err := organisationStore.CreateMember(r.Context(), invite); err != nil {
+					return err
+				}
+
+				loginSession, err := sesh.Login(sessionStore, r)
+				if err != nil {
+					return page.Paths.Supporter.Start.Redirect(w, r, appData)
+				}
+
+				loginSession.OrganisationID = invite.OrganisationID
```

Comment:
should also `loginSession.OrganisationName = invite.OrganisationName`, or get the organisation to use the current name